### ISSUE TITLE
vcpkg binary is always being rebuilt

### ISSUE THREAD

**Issue by vadi2** [lukka/run-vcpkg#77]
The newly-added "check if binary works" check seems to always be failing, despite the binary being cached. Across macOS/Linux/Windows builds, the check is always returning 1 and the binary is already being rebuilt:

```
2021-02-22T07:00:13.3308338Z ##[group]Run lukka/run-vcpkg@v7
2021-02-22T07:00:13.3308848Z with:
2021-02-22T07:00:13.3309774Z   vcpkgArguments: @/home/runner/work/Mudlet/Mudlet/3rdparty/our-vcpkg-dependencies/vcpkg-x64-linux-dependencies
2021-02-22T07:00:13.3310901Z   vcpkgDirectory: /home/runner/work/Mudlet/Mudlet/3rdparty/vcpkg
2021-02-22T07:00:13.3312164Z   appendedCacheKey: f180e0c15f8968099cfcd4e5d794450e7f752a92b056726ec61f8410bbe7402d-cachekey
2021-02-22T07:00:13.3313246Z   setupOnly: false
2021-02-22T07:00:13.3314194Z   vcpkgArtifactIgnoreEntries: !.git
**/*
!installed
!/vcpkg
!vcpkg.exe
!vcpkgArtifactIgnoreEntries

2021-02-22T07:00:13.3315361Z   vcpkgGitURL: https://github.com/microsoft/vcpkg.git
2021-02-22T07:00:13.3316043Z   cleanAfterBuild: true
2021-02-22T07:00:13.3316565Z   doNotUpdateVcpkg: false
2021-02-22T07:00:13.3317045Z   useShell: true
2021-02-22T07:00:13.3317469Z   doNotCache: false
2021-02-22T07:00:13.3317923Z env:
2021-02-22T07:00:13.3318428Z   BOOST_ROOT: /home/runner/work/Mudlet/Mudlet/3rdparty/boost
2021-02-22T07:00:13.3319211Z   BOOST_URL: https://sourceforge.net/projects/boost/files/boost/1.72.0/boost_1_72_0.tar.bz2/download
2021-02-22T07:00:13.3320207Z   pythonLocation: /opt/hostedtoolcache/Python/3.9.1/x64
2021-02-22T07:00:13.3321049Z   LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.9.1/x64/lib:/home/runner/work/Mudlet/Qt/5.14.1/gcc_64/lib
2021-02-22T07:00:13.3321889Z   PKG_CONFIG_PATH: /home/runner/work/Mudlet/Qt/5.14.1/gcc_64/lib/pkgconfig
2021-02-22T07:00:13.3322537Z   Qt5_Dir: /home/runner/work/Mudlet/Qt/5.14.1/gcc_64
2021-02-22T07:00:13.3323083Z   Qt5_DIR: /home/runner/work/Mudlet/Qt/5.14.1/gcc_64
2021-02-22T07:00:13.3323702Z   QT_PLUGIN_PATH: /home/runner/work/Mudlet/Qt/5.14.1/gcc_64/plugins
2021-02-22T07:00:13.3324369Z   QML2_IMPORT_PATH: /home/runner/work/Mudlet/Qt/5.14.1/gcc_64/qml
2021-02-22T07:00:13.3325461Z   vcpkgResponseFile: /home/runner/work/Mudlet/Mudlet/3rdparty/our-vcpkg-dependencies/vcpkg-x64-linux-dependencies
2021-02-22T07:00:13.3326440Z ##[endgroup]
2021-02-22T07:00:13.4244698Z vcpkg identified at commitId='4b222f8d2653603b74fa5086cf0b02a1963d3911', adding it to the cache's key.
2021-02-22T07:00:13.4247022Z Cache's key = 'submodGitId=4b222f8d2653603b74fa5086cf0b02a1963d3911-args=-1400687354-os=-2073516187-imageVer=-2108618863-appendedKey=935998087'.
2021-02-22T07:00:13.4248261Z ##[group]Restore vcpkg and its artifacts from cache
2021-02-22T07:00:13.4253688Z Cache's key = 'submodGitId=4b222f8d2653603b74fa5086cf0b02a1963d3911-args=-1400687354-os=-2073516187-imageVer=-2108618863-appendedKey=935998087', paths = '/home/runner/work/Mudlet/Mudlet/3rdparty/vcpkg,!/home/runner/work/Mudlet/Mudlet/3rdparty/vcpkg/packages,!/home/runner/work/Mudlet/Mudlet/3rdparty/vcpkg/buildtrees,!/home/runner/work/Mudlet/Mudlet/3rdparty/vcpkg/downloads'
2021-02-22T07:00:13.4255957Z Running restore-cache...
2021-02-22T07:00:14.1419393Z Cache Size: ~52 MB (54875019 B)
2021-02-22T07:00:14.1448024Z [command]/usr/bin/tar --use-compress-program zstd -d -xf /home/runner/work/_temp/d688779c-7b7f-4dd3-9d84-638e52f9df88/cache.tzst -P -C /home/runner/work/Mudlet/Mudlet
2021-02-22T07:00:15.5037748Z Cache hit, id=submodGitId=4b222f8d2653603b74fa5086cf0b02a1963d3911-args=-1400687354-os=-2073516187-imageVer=-2108618863-appendedKey=935998087.
2021-02-22T07:00:15.5041263Z ##[endgroup]
2021-02-22T07:00:15.5042002Z ⏱ elapsed: 2.076 seconds
2021-02-22T07:00:15.5045212Z ##[group]Set output env vars
2021-02-22T07:00:15.5046887Z Set the output variable 'RUNVCPKG_VCPKG_ROOT_OUT' to value: /home/runner/work/Mudlet/Mudlet/3rdparty/vcpkg
2021-02-22T07:00:15.5049453Z Set the environment variable 'RUNVCPKG_VCPKG_TRIPLET' to value: 
2021-02-22T07:00:15.5050445Z Set the output variable 'RUNVCPKG_VCPKG_TRIPLET_OUT' to value: 
2021-02-22T07:00:15.5051123Z ##[endgroup]
2021-02-22T07:00:15.5051764Z ⏱ elapsed: 0.001 seconds
2021-02-22T07:00:15.5178920Z tool: /usr/bin/git
2021-02-22T07:00:15.5180443Z Fetching the Git commit id at '/home/runner/work/Mudlet/Mudlet/3rdparty/vcpkg' ...
2021-02-22T07:00:15.5244477Z 4b222f8d2653603b74fa5086cf0b02a1963d3911
2021-02-22T07:00:15.5246870Z ##[group]Check whether vcpkg repository is up to date
2021-02-22T07:00:15.5248099Z Checking whether vcpkg's repository is updated to commit id '4b222f8d2653603b74fa5086cf0b02a1963d3911' ...
2021-02-22T07:00:15.5263084Z tool: /usr/bin/git
2021-02-22T07:00:15.6383099Z  4b222f8d2653603b74fa5086cf0b02a1963d3911 3rdparty/vcpkg (2020.04-1517-g4b222f8d2)
2021-02-22T07:00:15.6511901Z 'vcpkg' is detected as a submodule, adding '.git' to the ignored entries in '.artifactignore' file (for excluding it from caching).
2021-02-22T07:00:15.6513900Z File '.artifactsignore' content: ''**/*', '!installed', '!/vcpkg', '!vcpkg.exe', '!vcpkgArtifactIgnoreEntries', '.git''
2021-02-22T07:00:15.6515118Z Is vcpkg repository updated? Yes
2021-02-22T07:00:15.6516157Z ##[endgroup]
2021-02-22T07:00:15.6517028Z ⏱ elapsed: 0.116 seconds
2021-02-22T07:00:15.6518191Z ##[group]Check whether last vcpkg's build is up to date with sources
2021-02-22T07:00:15.6519665Z Checking last vcpkg build commit id in file '/home/runner/work/Mudlet/Mudlet/3rdparty/vcpkg/vcpkgLastBuiltCommitId' ...
2021-02-22T07:00:15.6521604Z vcpkg executable is up to date with sources.
2021-02-22T07:00:15.6522732Z ##[endgroup]
2021-02-22T07:00:15.6523773Z ⏱ elapsed: 0.006 seconds
2021-02-22T07:00:15.6524438Z ##[group]Check vcpkg executable exists
2021-02-22T07:00:15.6541324Z vcpkg executable exists at: '/home/runner/work/Mudlet/Mudlet/3rdparty/vcpkg/vcpkg'.
2021-02-22T07:00:16.6753354Z vcpkg executable returned code 1, forcing a rebuild.
2021-02-22T07:00:16.6754910Z ##[endgroup]
2021-02-22T07:00:16.6756353Z ⏱ elapsed: 1.029 seconds
2021-02-22T07:00:16.6757470Z ##[group]Build vcpkg
2021-02-22T07:00:16.6778738Z tool: /usr/bin/sh
2021-02-22T07:00:19.9989669Z -- The CXX compiler identification is GNU 10.2.0
```

**Comment by lukka** [comment]
@vadi2 that's right, i fixed it (in the depending packages) and fixed the unit test as well so it won't regress. I'll rebuild the action based on the fixed packages soon.

**Comment by lukka** [comment]
Released [v7.1](https://github.com/lukka/run-vcpkg/releases/tag/v7.1).

**Comment by vadi2** [comment]
Thanks lukka!
